internal/util/util.go

```diff
@@ -112,3 +113,18 @@ func WrapString(s string, lim uint) string {
 
 	return buf.String()
 }
+
+
+// ReadLine will read a single line from a bufio.Reader
+func ReadLine(r *bufio.Reader) (string, error) {
+	var (
+		isPrefix bool
+		err error
+		line, ln []byte
+	)
+	for isPrefix && err == nil {
+		line, isPrefix, err = r.ReadLine()
+		ln = append(ln, line...)
```

Comment:
Why are you using `line...` here? What is the effect?

Reply:
Because `line` is a byte array and I unpack it before appending it to the `ln` byte array.

Reply:
So, here you're basically trying the deal with the fact that we have very long lines, correct? Can you perhaps document this as the reason why we need to wrap ReadLine? (Otherwise a naive person reading the code could wonder why we're wrapping ReadLine)

Reply:
I suppose, I can do that though the docs of `bufio.Reader.ReadLine` pretty clearly say:

> ReadLine tries to return a single line, not including the end-of-line bytes. If the line was too long for the buffer then isPrefix is set and the beginning of the line is returned. 

How about I rename the function to `ReadLongLine`?

Reply:
Yes, that part of the docs is indeed why I guessed we were in such case. I am not necessarily confused by the function name, but I'd like to see the _reason_ why we expect long lines to be documented somewhere.

Reply:
> I'd like to see the reason why we expect long lines to be documented somewhere

I mean we should be prepared for anything. Tbh I find it a major oversight from the golang stdlib developers to make a functions that is called `ReadLine` and doesn't actually do what it's suppose to do.

How about I add a docstring to the docstring: "Just like the golang ReadLine, except it actually does what it should do.".

I would actually argue that the golang stdlib function should be called `ReadLineOrAPartOfIt` :P

Reply:
It seems to me we're using a deprecated interface [1]. Have you perhaps checked whether we could use bufio.Scanner [2] or bufio.Reader.ReadString [3] instead? (Both solutions seem to overcome the issue that ReadLine is bounded to a fixed size slice.)

[1] https://twitter.com/davecheney/status/604837853344989184?lang=en

[2] https://golang.org/pkg/bufio/#Scanner

[3] https://golang.org/pkg/bufio/#Reader.ReadString

Reply:
https://github.com/ooni/probe-cli/issues/57